### PR TITLE
Add a createPages snippet to GraphiQL explorer

### PR DESCRIPTION
**Pull request by graysonhicks**
## Description

This adds a `createPage` snippet to the GraphiQL explorer.  Often, a developer will begin creating the site by examining their data layer in GraphiQL.  They will then want to create pages based off of their initial query.  For example:

```graphql
query MyQuery {
  allContentfulBlogPosts {
    nodes {
        id
        slug
      }
    }
}

```

Currently, the user must then either:
- find the documentation on creating pages in gatsby-node, copy, paste and replace or
- type the createPage functionality by hand
This snippet allows for a direct copy and paste from GraphiQL.  The user will then only have to fill in:

- the path to the page template
- the field being used as the slug
- manually add desired fields to pageContext 

The above query would generate the following snippet:

```javascript
const path = require(`path`)

exports.createPages = async ({ graphql, actions }) => {
  const { createPage } = actions

  const result = await graphql(`
    {
      allContentfulBlogPosts {
          nodes {
            id
            slug
          }
        }
    }
  `)
  const templatePath = path.resolve(\`PATH/TO/TEMPLATE.js\`)

  result.data.allContentfulBlogPosts.nodes.forEach(( node ) => {
    createPage({
      path: NODE_SLUG,
      component: templatePath,
      context: {
        slug: NODE_SLUG,
      },
    })
  })
}

```

### Documentation

Snippets are not currently documented anywhere, but could be.  This one in particular could also be added in the createPages documentation in a callout

## Related Issues

None

### CODE REVIEW

File: packages/gatsby-graphiql-explorer/src/app/snippets.js [hunk 1]
```diff
@@ -69,4 +69,34 @@ export default ComponentName
...
+${getQuery(arg, 4)}
+  \`)
+
+  result.data.${
+    arg.operationDataList[0].operationDefinition.selectionSet.selections[0].name
+      .value
+  }.edges.forEach(({ node }) => {
+    createPage({
+      path: NODE_SLUG,
+      component: path.resolve(\`PATH/TO/TEMPLATE.js\`),
```
Comment (KyleAMathews) [inline packages/gatsby-graphiql-explorer/src/app/snippets.js:92]: It's a little thing but sometimes matters for perf but if you could move the path.resolve out of the forEach so it's only called once.

File: packages/gatsby-graphiql-explorer/src/app/snippets.js [hunk 2]
```diff
@@ -69,4 +69,34 @@ export default ComponentName
...
+  const { createPage } = actions
+
+  const result = await graphql(\`
+${getQuery(arg, 4)}
+  \`)
+
+  result.data.${
+    arg.operationDataList[0].operationDefinition.selectionSet.selections[0].name
+      .value
+  }.edges.forEach(({ node }) => {
```
Comment (LekoArts) [inline packages/gatsby-graphiql-explorer/src/app/snippets.js:89]: We don't really teach people the `edges { node }` pattern anymore and let them use `nodes` directly

Comment (graysonhicks) [reply packages/gatsby-graphiql-explorer/src/app/snippets.js]: Updated

File: packages/gatsby-graphiql-explorer/src/app/snippets.js (hunk 1 above)
Comment (graysonhicks) [reply packages/gatsby-graphiql-explorer/src/app/snippets.js]: Updated

**Review by vladar — APPROVED** [review]
Nice! 👍 🚢

**Pull request merged**